src/layouts/views/widget/index.js

```diff
@@ -189,21 +217,28 @@ class Widget extends Component {
     const { classes, ...other } = this.props;
 
     const {
-      isFail, loading, largeGrid, smallGrid, appExists,
+      loading, largeGrid, smallGrid, appExists, responseCode, submitAttempts,
     } = this.state;
 
+    if (loading) { return (<Loader />); }
+
     if (exists || appExists) {
       return <AppExistsView classes={classes} callbacks={{ onExists }} />;
     }
     if (errorMessage) {
       return <ErrorView classes={classes} callbacks={{ onFail }} />;
     }
-    if (isFail) {
-      return <FailError classes={classes} callbacks={{ onFail, onSubmit: this.submitData }} />;
+    if (responseCode !== 200) {
+      return (
+        <FailError
+          submitAttempts={submitAttempts}
+          classes={classes}
+          responseCode={responseCode}
+          callbacks={{ onFail, onSubmit: this.submitData }}
```

Comment:
is null a correct value for the responseCode? if a timeout occurs then the value will be null